django/db/migrations/operations/fields.py

```diff
@@ -24,10 +26,43 @@ def is_same_field_operation(self, operation):
         return self.is_same_model_operation(operation) and self.name_lower == operation.name_lower
 
     def references_model(self, name, app_label=None):
-        return name.lower() == self.model_name_lower
+        name_lower = name.lower()
+        if name_lower == self.model_name_lower:
+            return True
+
+        if self.field:
+            return field_references_model(self.field, ModelTuple(app_label, name_lower))
+
+        # Refuse the temptation to guess. This operation could be performed
+        # on a field referencing the specified model.
+        return True
 
     def references_field(self, model_name, name, app_label=None):
-        return self.references_model(model_name) and name.lower() == self.name_lower
+        model_name_lower = model_name.lower()
+        if model_name_lower == self.model_name_lower:
+            if name == self.name:
+                return True
+            elif self.field and hasattr(self.field, 'from_fields') and name in self.field.from_fields:
+                return True
+
+        if self.field:
+            model_tuple = ModelTuple(app_label, model_name_lower)
+            remote_field = self.field.remote_field
+            if remote_field:
```

Comment:
You've also written a `field_references_model` helper in the `utils` package that has some similar looking code. Is it possible that we could use that helper in here to simplify the code?